icu4j/main/classes/core/src/com/ibm/icu/impl/units/MeasureUnitImpl.java

```diff
@@ -70,7 +70,9 @@ public MeasureUnitImpl copy() {
         MeasureUnitImpl result = new MeasureUnitImpl();
         result.complexity = this.complexity;
         result.identifier = this.identifier;
-        result.singleUnits = new ArrayList<>(this.singleUnits);
+        for (SingleUnitImpl single : this.singleUnits) {
```

Comment:
Good catch!  I guess this is a Java-only change?

Reply:
In C++, MeasureUnitImpl::copy() has this in a for-loop, making use of the copy-constructor:

    SingleUnitImpl *item = result.units.emplaceBack(*units[i]);